hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java

```diff
@@ -55,34 +56,46 @@
      * @param hazelcastProperties array of {@link HazelcastProperty} to configure
      */
     protected void initProperties(Properties properties, HazelcastProperty[] hazelcastProperties) {
+        this.properties = properties;
         for (HazelcastProperty property : hazelcastProperties) {
```

Comment:
I'm working on a proper cleanup PR for the HazelcastProperties. There is some complexity here in the constructor and methods that need to be implemented that can be removed by simplification.